### DISCUSSION TITLE
Why no components per app?

### DISCUSSION THREAD

**Discussion by mjmare** [django-components/django-components#375] [Q&A] [answered]
From my reading of the docs it seems that components are separate from apps. Although that makes sense for general components  (buttons, banners etc), per app components seem useful as well. A bit like global and per app templates.

**Comment by EmilStenstrom** [comment]
We do support components per app. If you create a components directory in an app it should automatically load. If not, that’s a bug.

**Comment by mjmare** [comment]
Oh, I'm sorry. I had not understood from the docs. 
If I don't use global components, can I simplify installation by not touching APP_DIRS and not changing the TEMPLATE.loaders?

**Comment by JuroOravec** [comment, marked as answer]
For anyone coming across this later on, components per app have been available until v0.85 (not inclusive) and then again from v0.100 (inclusive).
- Until v0.85, the app-level components had to be imported manually inside `[app]/components.py` or `[app]/components/__init__.py` to work.

	E.g.
	```py
	import myapp.components.my_comp
	import myapp.components.my_table
	...
	```
- Since v0.100, the `[app]/components` are expected to be directories, where the python files are automatically loaded.
	- And since v0.100 you can also change / add which app-level directories are searched with the [`app_dirs` setting](https://github.com/EmilStenstrom/django-components#app_dirs)

---

As per this question

> If I don't use global components, can I simplify installation by not touching APP_DIRS and not changing the TEMPLATE.loaders?

@mjmare As of v0.100, theoretically if you are using only the inlined template, then yes. Otherwise not, because the TEMPLATE.loaders is required for Django to be able to load the HTML template as a Django Template object.

And by inlined template, I mean this:
```py
class MyComp(Component):
	template = """
		<div>
			<h1>
				{{ my_var }}
			</h1>
		</div>
	"""
```
